include/fmt/core.h

```diff
@@ -32,10 +32,15 @@
 #  define FMT_GCC_PRAGMA(arg)
 #endif
 
-#if defined(__INTEL_COMPILER)
+#ifdef __ICL
+#  define FMT_ICC_VERSION __ICL
+#  define FMT_ICC_POSIX 0
+#elif defined(__INTEL_COMPILER)
 #  define FMT_ICC_VERSION __INTEL_COMPILER
+#  define FMT_ICC_POSIX 1
 #else
 #  define FMT_ICC_VERSION 0
+#  define FMT_ICC_POSIX 1
```

Comment:
Why is ```FMT_ICC_POSIX``` equal to 1 for other compilers?

Reply:
You are making a good point. It is misleading. But it makes sure the intrinsic detection still works here for other compilers:
https://github.com/fmtlib/fmt/blob/17fb5de4a1fd1cfd4d93b8233c1b5cc03a43f2ef/include/fmt/format.h#L168-L169
Since this is essentially a work around for a compiler bug, perhaps we should introduce something like FMT_ICC_BUILTIN_CTZ_BUG to be crystal clear about it?